### PR TITLE
Update documentation for mxnet.metric.create

### PR DESCRIPTION
**Pull request by indhub**
@mli, @zackchase, @madjam, @nswamy, @Roshrini, @jiajiechen
Please review and merge.

### CODE REVIEW

File: python/mxnet/metric.py [hunk 1]
```diff
@@ -437,13 +437,29 @@ def feval(label, pred):
...
+        - Name of a metric.
+        - An instance of `EvalMetric`.
+        - A list, each element of which is a metric or a metric name.
+        - An evaluation function that computes custom metric for a given batch of
+          labels and predictions.
+
+    Examples
+    --------
+    >>> def custom_metric(label, pred):
+    ...     return np.mean(np.abs(label-pred))
```
Comment (madjam) [inline python/mxnet/metric.py]: Can you add a space before and after '-' for clarity.

Comment (indhub) [inline python/mxnet/metric.py]: Done.

File: python/mxnet/metric.py [hunk 2]
```diff
@@ -437,13 +437,29 @@ def feval(label, pred):
 # pylint: enable=invalid-name
 
 def create(metric, **kwargs):
-    """Create an evaluation metric.
+    """Creates an evaluation metric from metric name(s) or instance(s) of EvalMetric
```
Comment (zackchase) [inline python/mxnet/metric.py]: Can't we just say "Creates evaluation metrics from metric names or instances of EvalMetric"

Comment (zackchase) [inline python/mxnet/metric.py]: This covers the simple case where only one metric is given

Comment (indhub) [inline python/mxnet/metric.py]: Fixed.

**Pull request merged**